src/datachain/lib/dc.py

```diff
@@ -951,6 +951,41 @@ def merge(
 
         return ds
 
+    def subtract(  # type: ignore[override]
+        self,
+        other: "DataChain",
+        on: Optional[Union[str, Sequence[str]]] = None,
```

Comment:
So, for the files it will be like:
```
diff = dc1.subtract(dc2, ["file.source" "file.name", "file.etag"])
```

QUestions:
1) Is it possible to handle use cases when file name do not match? Like: `"file.name" == "wds.laion.file.name" AND "file.etag" == "wds.laion.file.etag"`

2) Is it possible to reuse the existing merge or join to get the diff/subtract?